File: src/main/java/maxdistructo/droidbot2/commands/Info.java
package maxdistructo.droidbot2.commands;

import java.util.List;
import java.util.Optional;

import maxdistructo.droidbot2.background.Config;
import maxdistructo.droidbot2.background.Perms;
import sx.blah.discord.handle.impl.obj.Message;
import sx.blah.discord.handle.obj.*;
import de.btobastian.sdcf4j.Command;
import de.btobastian.sdcf4j.CommandExecutor;

public class Info implements CommandExecutor{
   // @Command(aliases = {"/i", "/info"}, description = "Gets your info. Mods can get info on any user.", usage = "/i|info @User#0000")
    public static String onInfoCommand(Object[] args, IMessage message, IUser mentioned){
        IUser author = message.getAuthor();
        IGuild guild = message.getGuild();
        String guildName = guild.getName();
        String nick = author.getNicknameForGuild(guild);
        List<IRole> roles = author.getRolesForGuild(guild);
        Object[] rolesArray = roles.toArray();
        String roleNames = "";
        int i = 1;
        while(i < rolesArray.length)
        {
            if(i == rolesArray.length - 1){
                IRole roleCheck = (IRole)rolesArray[i];
                roleNames = roleNames + roleCheck.getName() + " ";
            }
            else {
                IRole roleCheck = (IRole) rolesArray[i];
                roleNames = roleNames + roleCheck.getName() + ", ";
            }
            i++;
        }
        if(mentioned == null){
            return "You are known as: " + nick + ". \nYou are in Discord Server: " + guildName + "\nYour roles in this server are: " + roleNames;
        }
        else if(Perms.checkMod(message)){
            IUser checked = mentioned;
            String nickChecked = checked.getNicknameForGuild(guild);
            List<IRole> rolesChecked = checked.getRolesForGuild(guild);
            Object[] rolesCheckedArray = rolesChecked.toArray();
            String roleNamesChecked = "";
            i = 1;
            while(i < rolesCheckedArray.length)
            {
                if(i == rolesCheckedArray.length - 1){
                    IRole roleCheck = (IRole)rolesCheckedArray[i];
                    roleNamesChecked = roleNamesChecked + roleCheck.getName() + " ";
                }
                else {
                    IRole roleCheck = (IRole) rolesCheckedArray[i];
                    roleNamesChecked = roleNamesChecked + roleCheck.getName() + ", ";
                }
                i++;
            }

            return "\nMember " + checked.getName() + " is also known as " + nickChecked + "\nThey have the roles " + roleNamesChecked + "\nIn Discord Server: " + guildName;

        }
        return "Command has errored. Please enter a valid command.";
    }
}

File: src/main/java/maxdistructo/discord/bots/droidbot/commands/Info.kt
package maxdistructo.discord.bots.droidbot.commands

import maxdistructo.discord.bots.droidbot.background.PrivUtils
import maxdistructo.discord.core.Perms
import maxdistructo.discord.core.Utils
import maxdistructo.discord.core.command.BaseCommand
import sx.blah.discord.handle.obj.IMessage
import sx.blah.discord.handle.obj.IRole
import sx.blah.discord.handle.obj.IUser


class Info : BaseCommand() {

    override val commandName: String
        get() = "info"
    override val helpMessage: String
        get() = "info - Gets your info in this server. Mods can mention a user to get information on another user"

    override fun init(message: IMessage, args: List<String>): String {
        return onInfoCommand(PrivUtils.listToArray(args), message, Utils.getMentionedUser(message))
    }

    fun onInfoCommand(args: Array<String>, message: IMessage, mentioned: IUser?): String {
        val author = message.author
        val guild = message.guild
        val guildName = guild.name
        val nick = author.getNicknameForGuild(guild)
        val roles = author.getRolesForGuild(guild)
        val rolesArray = roles.toTypedArray()
        var roleNames = ""
        var i = 1
        while (i < rolesArray.size) {
            if (i == rolesArray.size - 1) {
                val roleCheck = rolesArray[i] as IRole
                roleNames = roleNames + roleCheck.name + " "
            } else {
                val roleCheck = rolesArray[i] as IRole
                roleNames = roleNames + roleCheck.name + ", "
            }
            i++
        }
        if (mentioned == null) {
            return "You are known as: $nick. \nYou are in Discord Server: $guildName\nYour roles in this server are: $roleNames"
        } else if (Perms.checkMod(message)) {
            val nickChecked = mentioned.getNicknameForGuild(guild)
            val rolesChecked = mentioned.getRolesForGuild(guild)
            val rolesCheckedArray = rolesChecked.toTypedArray()
            var roleNamesChecked = ""
            i = 1
            while (i < rolesCheckedArray.size) {
                if (i == rolesCheckedArray.size - 1) {
                    val roleCheck = rolesCheckedArray[i] as IRole
                    roleNamesChecked = roleNamesChecked + roleCheck.name + " "
                } else {
                    val roleCheck = rolesCheckedArray[i] as IRole
                    roleNamesChecked = roleNamesChecked + roleCheck.name + ", "
                }
                i++
            }

            return "\nMember " + mentioned.name + " is also known as " + nickChecked + "\nThey have the roles " + roleNamesChecked + "\nIn Discord Server: " + guildName

        }
        return "Command has errored. Please enter a valid command."
    }
}
